src/communication-security/grpc-code/grpc_server_secured/server.go

```diff
@@ -0,0 +1,51 @@
+package main
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"log"
+	"net"
+
+	pb "pentest/grpc/samplebuff"
+
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials"
+)
+
+var (
+	port = flag.Int("port", 10001, "The server port")
+)
+
+// server is used to implement sample.GreeterServer.
+type server struct {
+	pb.UnimplementedSampleServiceServer
+}
```

Comment:
nit: I would suggest not embedding gRPC service interface to a structure like this. The issue is that this implicitly says the `server` struct implements the interface, and the program compiles with no issues. Now imaging we remove `server.Greet` method. The program should still compile, but now it panics in runtime when `Greet` is called. (it will panic because of nil receiver)

Instead, we can enforce interface implementation check on the compile time like this.

```suggestion
// Verify interface implementation on compile.
var _ pb.UnimplementedSampleServiceServer = (*server)(nil)

// server implements sample.GreeterServer interface.
type server struct {}
```

Note, the interface implementation check will be implicitly performed by compiler by the `pb.RegisterSampleServiceServer(s, &server{})` call below. This is ok but explicit check with `var _ ...` is more obvious for a reader.